src/components/FitExtent/FitExtent.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FaExpand } from 'react-icons/fa';
+import OLMap from 'ol/Map';
+import Button from '../Button';
+
+const propTypes = {
+  /**
+   * An ol map.
+   */
+  map: PropTypes.instanceOf(OLMap).isRequired,
+
+  /**
+   * The extent to be zoomed.
+   */
+  extent: PropTypes.arrayOf(PropTypes.number).isRequired,
+
+  /**
+   * Title for the fitExtent button.
+   */
+  title: PropTypes.string,
+
+  /**
+   * CSS class of the fitExtent button.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Children content of the fitExtent button.
+   */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  title: 'Fit Extent',
+  className: 'tm-button tm-round-blue',
+  children: <FaExpand focusable={false} />,
+};
+
+const fitExtentFc = (map, extent) => {
```

Comment:
for keeping it simple, please remove this function and directly perform this actions on click.